kpatch-build/kpatch-elf.c

```diff
@@ -328,9 +353,32 @@ void kpatch_create_symbol_list(struct kpatch_elf *kelf)
 					sym->name);
 
 			if (is_bundleable(sym)) {
-				if (sym->sym.st_value != 0)
-					ERROR("symbol %s at offset %lu within section %s, expected 0",
+				/*
+				 * function prologue generated by gcc6 on ppc64le has
+				 * the sequence:
+				 *
+				 *	.globl my_func
+				 *	.type my_func, @function
+				 *	.quad .TOC.-my_func
+				 * my_func:
+				 *	.reloc ., R_PPC64_ENTRY ; optional
+				 *	ld r2,-8(r12)
+				 *	add r2,r2,r12
+				 * 	.localentry my_func, .-my_func
+				 *
+				 * where .localentry for the functions, will generated
+				 * st_value of 0x8 for the local functions. Skip, this
+				 * case.
+				 */
+				if (sym->sym.st_value != 0) {
+#ifdef __powerpc__
+					if (!is_local_func(sym) ||
+					    !has_local_entry_offset(sym->sym) ||
+					    (sym->sym.st_value != 0x8))
+#endif
+						ERROR("symbol %s at offset %lu within section %s, expected 0",
 					      sym->name, sym->sym.st_value, sym->sec->name);
+				}
 				sym->sec->sym = sym;
```

Comment:
I propose the following changes to make this code clearer:

- Move the long comment to above the `has_local_entry_offset()` function, since it describes the intent of that function.  It's too much information for this location, IMO.
- Remove the "Skip, this case" comment because nothing is skipped.
- Rename `has_local_entry_offset()` to `is_localentry_sym()` and do the `is_local_func()` and `st_value == 0x8` checks inside that function.
- Add a newline between the error checking clause and the `sym->sec->sym = sym;`.  Otherwise the most important line of code is visually obscured.